test/integration/test/philips_light_test.js

```diff
@@ -38,57 +35,117 @@ Prepper.makeSuite('Control lights locally',function(){
+    // id and the philips hue id until the tag feature is implemented
+    it('Turn on all lights one by one', function(){
+      var setURL = channelURL + '/set';
+      var lightID = lightKeys[0].replace('getter','setter');
+      var payload = {'select': {'id': lightID}, 'value': { 'OnOff': 'On' } };
+      
+      return chakram.put(setURL,payload)
+      .then(function(cmdResponse) {
+       expect(cmdResponse).to.have.status(200);
+       expect(Object.keys(cmdResponse.body)[0]).equals(lightID);
+       lightID = lightKeys[1].replace('getter','setter');
+       payload = {'select': {'id': lightID}, 'value': { 'OnOff': 'On' } };
+       return chakram.put(setURL,payload);
+     })
+      .then(function(cmdResponse) {
+        expect(cmdResponse).to.have.status(200);
+        expect(Object.keys(cmdResponse.body)[0]).equals(lightID);
+        lightID = lightKeys[2].replace('getter','setter');
+        payload = {'select': {'id': lightID}, 'value': { 'OnOff': 'On' } };
+        return chakram.put(setURL,payload);
+      })
+      .then(function(cmdResponse) {
+        expect(cmdResponse).to.have.status(200);
+        expect(Object.keys(cmdResponse.body)[0]).equals(lightID);
+        // check no lights are turned off now
+        expect(Prepper.philipshue_server.lightStatus(1) &&
+        Prepper.philipshue_server.lightStatus(2) &&
+        Prepper.philipshue_server.lightStatus(3)).equals(true);
+
+        //check all lights are reported to be on
+        var getURL = channelURL + '/get';
+        var getterPayload = [{'kind': 'LightOn'}];  
+        return chakram.put(getURL,getterPayload);
+      })
+      .then(function(listResponse) {
+        lightKeys = Object.keys(listResponse.body);
+        expect(lightKeys.length).equals(3);
+        expect(listResponse).to.have.status(200);
 
-      // Currently, there is no mapping between the foxbox
-      // issues id and the philips hue id until the tag feature is implemented
-      it('Turn on all lights', function(){
-        return Prepper.turnOnLight(lightinfo, serviceURL, 0)
-        .then(function(cmdResponse) {
-          Prepper.expectLightIsOn(cmdResponse);
-          return Prepper.turnOnLight(lightinfo, serviceURL, 1);
-        })
-        .then(function(cmdResponse) {
-          Prepper.expectLightIsOn(cmdResponse);
-          return Prepper.turnOnLight(lightinfo, serviceURL, 2);
-        })
-        .then(function(cmdResponse) {
-         Prepper.expectLightIsOn(cmdResponse);
-
-         // check no lights are turned off now
-         expect(Prepper.philipshue_server.lightStatus(1) &&
-          Prepper.philipshue_server.lightStatus(2) &&
-          Prepper.philipshue_server.lightStatus(3)).equals(true);
-       });
-      });
-});
-});
+        for (var i =0; i < lightKeys.length; i++){
+          expect(listResponse.body[lightKeys[i]].OnOff).equals('On');  
+        }
+      }); 
+    });
+
+    it('Turn off all lights at once', function() {
```

Comment:
Actually, it doesn't.  it'll turn off (if it is off, send the off command again) so it doesn't 'depend' on the previous test, although having it turned on before would make the test more meaningful. But I'm thinking even sending the off command to the already off test has some merit.